### DISCUSSION TITLE
Overlaps in 2d hpmc simulation for mixture of ellipses and circles.

### DISCUSSION THREAD

**Discussion by sauravIISc** [glotzerlab/hoomd-blue#1759] [Support] [answered]
Hi,

I am trying to simulate a mixture of circles and ellipses in a 2D box using `hoomd.hpmc.integrate.Ellipsoid`. In 2D, after a short preliminary run, I see overlaps between particles when I visualize them. However, `mc.overlaps` from the integrator gives me a value of 0. We know overlaps are prohibited for hard particles, and I see overlap when I visualize them. I suspect it could be due to some bugs in the integrator when simulations are performed in 2d because there were no overlaps when I tried the 3d box. I could also be making some mistakes in defining the parameters, so any help or clarification is appreciated. I have also attached a minimal script to see the overlaps.
[minimal_script.zip](https://github.com/glotzerlab/hoomd-blue/files/14892751/minimal_script.zip)

Thanks in advance,

**Comment by joaander** [comment]
You closed this discussion. Were you able to resolve the issue?

**Reply by sauravIISc** [reply]
Thank you for the reply. I solved the issue temporarily by defining a nonzero dimension (c) for the ellipse and circles in the z-direction, keeping everything else the same.

**Reply by joaander** [reply, marked as answer]
Yes, this is the workaround you should use. The fast numerical methods HOOMD-blue uses to detect overlaps assume that `a`, `b` and `c` are all non-zero. At the same time, your shapes will only rotate about the z axis in 2D boxes *effectively* giving you ellipses as desired.

I opened #1762 so that a future HOOMD-blue release will provide an error message when given `Ellipsoid` is given a 0 value.

**Reply by sauravIISc** [reply]
Thanks, it clarified the issue.
